packages/aws-cdk-lib/aws-eks/lib/managed-nodegroup.ts

```diff
@@ -476,7 +476,7 @@ export class Nodegroup extends Resource implements INodegroup {
       resource: 'nodegroup',
       resourceName: this.physicalName,
     });
-    this.nodegroupName = this.getResourceNameAttribute(resource.ref);
+    this.nodegroupName = resource.getAtt('NodegroupName').toString();
```

Comment:
```suggestion
    this.nodegroupName = this.getResourceNameAttribute(resource.attrNodegroupName);
```

Can we do this instead?